### PR TITLE
Rework env variable processing

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: lib/src/main/java/org/sonarsource/scanner/lib/EnvironmentConfig.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  public static Map<String, String> load(LogOutput logger) {
+    return load(System.getenv(), logger);
+  }
+
+  static Map<String, String> load(Map<String, String> env, LogOutput logger) {
+    var loadedProps = new HashMap<String, String>();
+    Optional.ofNullable(env.get(SONAR_HOST_URL_ENV_VAR)).ifPresent(url -> loadedProps.put(ScannerProperties.HOST_URL, url));
+    Optional.ofNullable(env.get(SONAR_USER_HOME_ENV_VAR)).ifPresent(path -> loadedProps.put(ScannerProperties.SONAR_USER_HOME, path));
```
Comment (ericg138) [inline lib/src/main/java/org/sonarsource/scanner/lib/EnvironmentConfig.java:51]: SONAR_LOGIN and SONAR_TOKEN will be handled here as well right? I suppose we'll add them when we add the authentication

File: lib/src/main/java/org/sonarsource/scanner/lib/Utils.java [hunk 2]
```diff
@@ -32,58 +28,14 @@
...
-      sb.append(it.next());
-      if (!it.hasNext()) {
-        break;
-      }
-      sb.append(delimiter);
-    }
-    return sb.toString();
-  }
-
   static void writeProperties(File outputFile, Properties p) {
```
Comment (ericg138) [inline lib/src/main/java/org/sonarsource/scanner/lib/Utils.java:39]: Not really related but I think we can remove this method too

**Review by ericg138 — APPROVED** [review]
LGTM

File: lib/src/main/java/org/sonarsource/scanner/lib/EnvironmentConfig.java (hunk 1 above)
Comment (henryju) [reply lib/src/main/java/org/sonarsource/scanner/lib/EnvironmentConfig.java:51]: Yes!

File: lib/src/main/java/org/sonarsource/scanner/lib/Utils.java (hunk 2 above)
Comment (henryju) [reply lib/src/main/java/org/sonarsource/scanner/lib/Utils.java:39]: I will do it in the next PR, good catch!

**Pull request merged**